api/src/main/java/org/apache/iceberg/expressions/Literals.java

```diff
@@ -88,6 +88,22 @@ static <T> Literal<T> from(T value) {
         "Cannot create expression literal from %s: %s", value.getClass().getName(), value));
   }
 
+  public static Literal<Object> fromJavaType(Object value) {
```

Comment:
I don't think this should be added to the public API. We avoided adding support for these because Iceberg is not responsible for time zone conversions.

It would be fine to revisit that decision, but I think this is only used so that the implementation is kept simple, although it will have poor performance. Instead, I think this should define real functions that are typed, instead of accepting any input type and dynamically calling functions.

Reply:
+1. Particularly about keeping this out of `api` where we have compatibility guarantees.

Reply:
Also calling `Literal.of(value.toString())` for every input seems like a performance concern.

Though I agree with Ryan that with some minor changes this should be a good addition overall :)

Reply:
Yes, this way is really not suitable. After the change, this function is no longer used. Deleted.